src/ucp/core/ucp_mm.inl

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+ *
+ * See file LICENSE for terms.
+ */
+
+#ifndef UCP_MM_INL_
+#define UCP_MM_INL_
+
+#include "ucp_mm.h"
+
+
+extern ucp_mem_h ucp_mem_dummy_handle;
+
+static UCS_F_ALWAYS_INLINE int
+ucp_memh_is_zero_length(const ucp_mem_h memh) {
+    return memh == ucp_mem_dummy_handle;
+}
+
+static UCS_F_ALWAYS_INLINE ucs_status_t
+ucp_memh_get(ucp_context_h context, void *address, size_t length,
+             ucs_memory_type_t mem_type, ucp_md_map_t reg_md_map,
+             unsigned uct_flags, ucp_mem_h *memh_p)
+{
+    ucs_rcache_region_t *rregion;
+    ucs_status_t status;
+    ucp_mem_h memh;
+
+    if (length == 0) {
+        *memh_p = ucp_mem_dummy_handle;
+        return UCS_OK;
+    }
+
+    if (context->rcache != NULL) {
+        status = ucs_rcache_get(context->rcache, address, length,
+                                PROT_READ|PROT_WRITE, NULL, &rregion);
```

Comment:
need to pass rot flags to ucp_memh_get from the relevant protocol
otherwise, we may fail to send with read-only memory